element_array_ephys/ephys_organoids.py

```diff
@@ -128,382 +129,185 @@ class AcquisitionSoftware(dj.Lookup):
+            del data, lfp
 
-            self.insert1(
-                dict(
-                    key,
-                    lfp_sampling_rate=spikeglx_recording.lfmeta.meta["imSampRate"],
-                    lfp_time_stamps=(
-                        np.arange(lfp.shape[1])
-                        / spikeglx_recording.lfmeta.meta["imSampRate"]
-                    ),
-                    lfp_mean=lfp.mean(axis=0),
-                )
-            )
-
-            electrode_query = (
-                probe.ProbeType.Electrode
-                * probe.ElectrodeConfig.Electrode
-                * EphysRecording
-                & key
-            )
-            probe_electrodes = {
-                (shank, shank_col, shank_row): key
-                for key, shank, shank_col, shank_row in zip(
-                    *electrode_query.fetch("KEY", "shank", "shank_col", "shank_row")
-                )
+        self.insert1(
+            {
+                **key,
+                "lfp_sampling_rate": TARGET_SAMPLING_RATE,
+                "header": header,
             }
+        )
 
-            for recorded_site in lfp_channel_ind:
-                shank, shank_col, shank_row, _ = spikeglx_recording.apmeta.shankmap[
-                    "data"
-                ][recorded_site]
-                electrode_keys.append(probe_electrodes[(shank, shank_col, shank_row)])
-        elif acq_software == "Open Ephys":
-            oe_probe = get_openephys_probe_data(key)
-
-            lfp_channel_ind = np.r_[
-                len(oe_probe.lfp_meta["channels_indices"])
-                - 1 : 0 : -self._skip_channel_counts
-            ]
-
-            # (sample x channel)
-            lfp = oe_probe.lfp_timeseries[:, lfp_channel_ind]
-            lfp = (
-                lfp * np.array(oe_probe.lfp_meta["channels_gains"])[lfp_channel_ind]
-            ).T  # (channel x sample)
-            lfp_timestamps = oe_probe.lfp_timestamps
+        electrode_query = EphysSession * probe.ElectrodeConfig.Electrode & key
 
-            self.insert1(
-                dict(
-                    key,
-                    lfp_sampling_rate=oe_probe.lfp_meta["sample_rate"],
-                    lfp_time_stamps=lfp_timestamps,
-                    lfp_mean=lfp.mean(axis=0),
-                )
+        # Single insert in loop to mitigate potential memory issue.
+        for ch, lfp in zip(channels, lfp_concat):
+            # Powerline noise removal
+            b_notch, a_notch = signal.iirnotch(
+                w0=powerline_noise_freq, Q=30, fs=TARGET_SAMPLING_RATE
             )
+            lfp = signal.filtfilt(b_notch, a_notch, lfp)
 
-            electrode_query = (
-                probe.ProbeType.Electrode
-                * probe.ElectrodeConfig.Electrode
-                * EphysRecording
-                & key
+            # Bandpass filter
+            b_butter, a_butter = signal.butter(
+                N=4, Wn=[0.5, 300], btype="bandpass", fs=TARGET_SAMPLING_RATE
```

Comment:
I suggest that we mirror the sampling rate and filtering frequencies of the Neuropixels probes:
```
The signals from each recording site are split in hardware into a spike band (30 kHz sampling rate, 500 Hz highpass filter) and an LFP band (2.5 kHz sampling rate, 1000 Hz lowpass filter). 
```
Source: [Allen Brain Observatory - Technical White Paper](https://brainmapportal-live-4cc80a57cd6e400d854-f7fdcae.divio-media.net/filer_public/80/75/8075a100-ca64-429a-b39a-569121b612b2/neuropixels_visual_coding_-_white_paper_v10.pdf)